### ISSUE TITLE
[Feature]: **Feature Request: Gateway for Model to Support Multiple Models Generation in a Given Context**

### ISSUE THREAD

**Issue by ronyadgar** [vllm-project/vllm#6906]
### 🚀 The feature, motivation and pitch

I would like to propose a feature that creates a gateway for a model, allowing for multiple models to generate content within a specific context. This is similar to the concept of speculative decoding, where it is sometimes unnecessary to query the model because the probability distribution is well-known.

This feature can dramatically accelerate token generation in scenarios where the next token is deterministic and does not require the usual forward pass. 

For example, let's assume that I want to generate a JSON completion according to a specific JSON schema. Clearly, the keys of the JSON are mandatory in the response, so at some points, it is obvious what the next token prediction will be.

### Alternatives

An alternative approach is to add a logit processor, which would mask the rest of the tokens in cases of deterministic tokens. However, this approach results in redundant processing to calculate the logits.

### Additional context

In my specific case, I would like the model to generate answers according to the unified diff format. Therefore, in many states, the next token is well-known, especially when the model needs to generate token quotes from the code.
